wpilib/wpilib/digitaloutput.py

```diff
@@ -1,71 +1,65 @@
-# validated: 2017-11-19 EN b65447b6f5a8 edu/wpi/first/wpilibj/DigitalOutput.java
-#----------------------------------------------------------------------------
+# validated: 2017-11-23 TW b65447b6f5a8 edu/wpi/first/wpilibj/DigitalOutput.java
+# ----------------------------------------------------------------------------
 # Copyright (c) FIRST 2008-2012. All Rights Reserved.
 # Open Source Software - may be modified and shared by FRC teams. The code
 # must be accompanied by the FIRST BSD license file in the root directory of
 # the project.
-#----------------------------------------------------------------------------
-
-import warnings
-import weakref
+# ----------------------------------------------------------------------------
 
 import hal
-
 from .digitalsource import DigitalSource
 from .sensorbase import SensorBase
 
 __all__ = ["DigitalOutput"]
 
+
 def _freePWMGenerator(pwmGenerator):
     # Disable the output by routing to a dead bit.
     hal.setDigitalPWMOutputChannel(pwmGenerator, SensorBase.kDigitalChannels)
     hal.freeDigitalPWM(pwmGenerator)
 
+
 class DigitalOutput(DigitalSource):
     """Writes to a digital output
     
     Other devices that are implemented elsewhere will automatically allocate
     digital inputs and outputs as required.
     """
 
+    invalidPwmGenerator = 0
```

Comment:
Note that this is a private constant upstream. I think we are perfectly fine with using `None` for this instead.